### PR TITLE
Make compatible with K8s 1.18

### PR DESCRIPTION
**Pull request by curamos**
# Kubernetes 1.18:
$ kubectl api-versions | grep networking
networking.k8s.io/v1
networking.k8s.io/v1beta1
$ kubectl explain --api-version=networking.k8s.io/v1 Ingress
error: Couldn't find resource for "networking.k8s.io/v1, Kind=Ingress"

### CODE REVIEW

**Comment by caiofralmeida** [comment]
@curamos When I did a test putting the resource name on the conditions, isn't worked for me. Because of that, I opened a PR that parameterizes the apiVersion via values https://github.com/oauth2-proxy/manifests/pull/41.

**Comment by morganchristiansson** [comment]
use chart version 3.3.2 as workaround

**Comment by caiofralmeida** [comment]
@morganchristiansson thanks, but I need to change the service account name, I opened this PR https://github.com/oauth2-proxy/manifests/pull/39 to fix it. I think this but will not fix it in chart version 3.3.2, right?

**Comment by pierluigilenoci** [comment]
I find this fix better than specifying the API version directly in the values file (ref: https://github.com/oauth2-proxy/manifests/pull/41).

Could you also bump the chart and squash all in a single commit? 

@desaintmartin what do you think about it?

File: helm/oauth2-proxy/templates/ingress.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 {{- $ingressPathType := .Values.ingress.pathType -}}
 {{- $extraPaths := .Values.ingress.extraPaths -}}
 {{- $apiV1 := false -}}
-{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1" -}}
+{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" -}}
 apiVersion: networking.k8s.io/v1
 {{- $apiV1 = true -}}
 {{- else if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1" -}}
```
Comment (pierluigilenoci) [inline helm/oauth2-proxy/templates/ingress.yaml:11]: Should `/Ingress` be specified also here?

**Comment by pierluigilenoci** [comment]
> @morganchristiansson thanks, but I need to change the service account name, I opened this PR #39 to fix it. I think this but will not fix it in chart version 3.3.2, right?

Of course not. 😄

File: helm/oauth2-proxy/templates/ingress.yaml (hunk 1 above)
Comment (pierluigilenoci) [reply helm/oauth2-proxy/templates/ingress.yaml:11]: I'd prefer to keep this PR as complete as possible. 😜 
For me, it makes sense to check inside the Ingress manifest if ".Capabilities.APIVersions.Has" for the Ingress specific API in the whole file, not only in one point.

**Comment by pierluigilenoci** [comment]
The chart needs another version bump.

File: helm/oauth2-proxy/templates/ingress.yaml (hunk 1 above)
Comment (curamos) [reply helm/oauth2-proxy/templates/ingress.yaml]: Changed it.

**Review by pierluigilenoci — APPROVED** [review]

**Pull request merged**